skllm/models/gpt_zero_shot_clf.py

```diff
@@ -162,11 +256,10 @@ def _predict_single(self, x):
             labels = []
 
         labels = list(filter(lambda l: l in self.classes_, labels))
-
-        if len(labels) > self.max_labels:
-            labels = labels[: self.max_labels - 1]
-        elif len(labels) < 1:
-            labels = [random.choices(self.classes_, self.probabilities_)[0]]
+        if len(labels) == 0:
+            labels = self._get_default_label()
+        if labels is not None and len(labels) > self.max_labels:
+            labels = random.choices(labels, k=self.max_labels)
```

Comment:
I would simply truncate a list as it was before. We assume that the entries in the list returned by GPT are sorted based on certainty. So, under this assumption, if the max number of labels is exceeded, it is better to take the ones from the beginning of the list. However, there is no quantitative evidence for that.

Reply:
changed